vignettes/web_only/BCRpipeline.Rmd

```diff
@@ -233,14 +239,81 @@ bcr$full_clones$Cluster %>% unique()
 We have found mismatches between a germline and an ancestor sequence. Dots represent nucleotides matches between the sequences, letters represent mismatches between the sequences:
 
 ```{r example 12}
+# the example of common ancestor sequence
+bcr$full_clones$Common.Ancestor[1]
+
+# the example of mismatches between a germline and an ancestor sequence
 bcr$full_clones$Germline.Output[1]
 ```
 
 We have calculated a trunk length for each cluster:
 
 ```{r example 13}
-bcr$full_clones$Trunk.Length
+bcr$full_clones[ , c('Cluster', 'Trunk.Length') ]
 ```
+
+# Somatic hypermutation analysis
+
+Somatic hypermutation analysis is extremely useful for BCRs. The somatic hypermutation rate allows us to estimate the maturation of repertoire and figure out what kind of mutation contributes to the emergence of high-affinity antibodies.
+
+In Immunarch, `repSomaticHypermutation()` function is designed for hypermutation analysis:
+
+```{r example 14, , warning = FALSE}
+bcr_data <- bcrdata$data
+
+shm_data <- bcr_data %>%
+  seqCluster(seqDist(bcr_data), .fixed_threshold = 3) %>%
+  repGermline(.threads = 2) %>%
+  repAlignLineage(.min_lineage_sequences = 2, .align_threads = 2, .nofail = TRUE) %>%
+  repClonalFamily(.threads = 2, .nofail = TRUE) %>%
+  repSomaticHypermutation(.threads = 2, .nofail = TRUE)
+```
+The function `repSomaticHypermutation()` takes germline V,  germline J sequences and clonotype V and clonotypes J sequences and alignes germline and clonotype sequences to calculate mutations.
```

Comment:
The function `repSomaticHypermutation()` takes V and J germline sequences and V and J clonotype sequences as an input. Then the function aligns germline and clonotype sequences to detect and calculate occurring mutations.